### PR TITLE
V2 syntax with some fixes in core

### PR DESCRIPTION
**Pull request by kell18**
- V2 syntax
- Adaptation of flink-connector project to core.v2
- Timestamp to Idx extractor (Pattern.scala)
- Some fixes in core

### CODE REVIEW

**Comment by fabura** [comment]
Doesn't compile for me:
[info] Done compiling.
[error] /Users/fbr/IdeaProjects/tsp/dsl/src/main/scala/ru/itclover/tsp/dsl/v2/ASTPatternGenerator.scala:33:63: type mismatch;
[error]  found   : T => ru.itclover.tsp.v2.Result[T]
[error]  required: Any => ru.itclover.tsp.v2.Result[?]
[error]               registry.getFunction1(fc.functionName).getOrElse(sys.error(s"Function ${fc.functionName} not found"))
[error]                                                               ^
[error] /Users/fbr/IdeaProjects/tsp/dsl/src/main/scala/ru/itclover/tsp/dsl/v2/ASTPatternGenerator.scala:37:63: type mismatch;
[error]  found   : (ru.itclover.tsp.v2.Result[T], ru.itclover.tsp.v2.Result[T]) => ru.itclover.tsp.v2.Result[Nothing]
[error]  required: (ru.itclover.tsp.v2.Result[Any], ru.itclover.tsp.v2.Result[Any]) => ru.itclover.tsp.v2.Result[?]
[error]               registry.getFunction2(fc.functionName).getOrElse(sys.error(s"Function ${fc.functionName} not found"))
[error]                                                               ^
[error] /Users/fbr/IdeaProjects/tsp/dsl/src/main/scala/ru/itclover/tsp/dsl/v2/ASTPatternGenerator.scala:45:63: type mismatch;
[error]  found   : (ru.itclover.tsp.v2.Result[Any], ru.itclover.tsp.v2.Result[T]) => ru.itclover.tsp.v2.Result[Any]
[error]  required: (ru.itclover.tsp.v2.Result[?], ru.itclover.tsp.v2.Result[Any]) => ru.itclover.tsp.v2.Result[?]
[error]         new ReducePattern(ffc.arguments.map(generatePattern))(func, ffc.cond, Result.succ(initial))
[error]                                                               ^

**Comment by hyst329** [comment]
@fabura `getFunction1` and `getFunction2` were in the (very) old version of this branch, so try to re-pull please (because these methods don't exist anymore)

**Comment by hyst329** [comment]
With [this](https://github.com/Clover-Group/tsp/pull/17/commits/c3db300bcca273b540174895b839edf33902855f) commit I've added chunking to the real `PatternFlatMapper`, since the previous implementation used a "chunk" of 1 event (the default chunk limit is now 1000, it should probably be adjustable through request options), but still doubting about the correctness (and thread safety, because the implementation uses mutable state, hope that each thread has its own `PatternFlatMapper`). Also, to avoid re-emitting of previous events (since the inner queues grow and don't clear for some reason), a set of processed indices are now stored in the PFM (so the events which have already processed are not re-emitted)

File: flink/src/main/scala/ru/itclover/tsp/mappers/ProcessorCombinator.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package ru.itclover.tsp.mappers
+import org.apache.flink.streaming.api.scala.function.ProcessWindowFunction
+import org.apache.flink.streaming.api.windowing.windows.Window
+import org.apache.flink.util.Collector
+import ru.itclover.tsp.mappers.PatternProcessor
+import ru.itclover.tsp.v2.PState
+
+import scala.reflect.ClassTag
+//
+//import org.apache.flink.api.common.functions.{AbstractRichFunction, RichFlatMapFunction, RuntimeContext}
```
Comment (kell18) [inline flink/src/main/scala/ru/itclover/tsp/mappers/ProcessorCombinator.scala]: Dead comments, no need to store it (it's in history)

File: flink/src/main/scala/ru/itclover/tsp/mappers/PatternProcessor.scala [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.flink.streaming.api.windowing.windows.Window
+import org.apache.flink.util.Collector
+import ru.itclover.tsp.io.TimeExtractor
+import ru.itclover.tsp.v2.Pattern.QI
+import ru.itclover.tsp.v2.{PState, Pattern, StateMachine, Succ}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ListBuffer
+
+//case class PatternFlatMapper[E, State <: PState[Inner, State], Inner, Out](
```
Comment (kell18) [inline flink/src/main/scala/ru/itclover/tsp/mappers/PatternProcessor.scala]: Dead comments, no need to store it (it's in history)

Comment (hyst329) [reply flink/src/main/scala/ru/itclover/tsp/mappers/PatternProcessor.scala]: Fixed

File: flink/src/main/scala/ru/itclover/tsp/mappers/ProcessorCombinator.scala (hunk 1 above)
Comment (hyst329) [reply flink/src/main/scala/ru/itclover/tsp/mappers/ProcessorCombinator.scala]: Fixed

**Comment by kell18** [comment]
Guys, let's close this pull request? It's far too much for one 🛑

**Comment by kell18** [comment]
Excep tests that check `1.json == 1` looks good to me :)

**Comment by tampler** [comment]
> It is basic Akka Spray-Json feature to serialize stuff like that, why we need to check it?
I didn't change anything, yet. Just removed old code. All tests passed. The coverage is +80% for all modules, but Core. If you're okay, confirm the merge

**Pull request merged**